microsoft/testsuites/performance/storageperf.py

```diff
@@ -455,13 +464,34 @@ def _perf_premium_datadisks(
         max_iodepth: int = 256,
     ) -> None:
         disk = node.features[Disk]
-        data_disks = disk.get_raw_data_disks()
-        disk_count = len(data_disks)
-        assert_that(disk_count).described_as(
-            "At least 1 data disk for fio testing."
-        ).is_greater_than(0)
-        partition_disks = reset_partitions(node, data_disks)
-        filename = ":".join(partition_disks)
+        if disk.disk_controller_type == 'nvme':
```

Comment:
how the nvme disk exist. do they exist for some vm sizes by default. no need to attach data disks?

Reply:
In VMs with disk controller type = nvme even non nvme disks are detected as nvme disks. They perform (IOPS, throughput) just like regular azure data disks but guest uses nvme driver to talk to them. This happens when ASAP is enabled. The current SKU which I am testing doesn't have NVMe disk support but as ASAP enabled, host presents all data disks to guest as PCI NVMe devices and NVMe driver is used to talk to all data disks including OS disk.

Reply:
this method will be used case which attaches many data disks, I don't think your scenario needs attach extra data disks? you need add a new case and add disk controller type = nvme as requirement.

Reply:
@LiliDeng ASAP SKUs (Ebs_v5) deploy with only OS disk like most VM SKUs. In order to test disk performance, we need attach many disks. This is done the exact same way as most SKUs. The only difference is that the disks will show up as NVMe devices even when they are SCSI disks.

From the test perspective, the only difference is the disk names. This does not require different requirements or a different test.